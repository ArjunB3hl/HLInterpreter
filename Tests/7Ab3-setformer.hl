// Test set formers
// Test the structure 
{i in {} | #1};
{i in {1} | #1};
{i in {1,2,3,4} | #1};
{i in {1,2,3,4} | #0};

// Test the conditions that depend on i
{j in {1,2,3,4,5,6,7,8,9,10} | j%3==0};
{i in [1,100] | i%11 == 0};

// Nest set formers in domain
{i in {j in {2,3,4,5} | #1} | i%2==0};
{i in {j in [1,100] | j%3==0} | i%2==0};
{i in {i in [1,100] | i%3==0} | i%2==0};

// Nest set formers in condition
{i in [1,100] | i =in {j in [1,100] | j%11==0} };
{i in [1,100] | i =in {j in {k in [1,100] | k%3==0} | j%5==0} };
{i in [1,100] | i =in {i in {i in [1,100] | i%3==0} | i%5==0} };

//Nest them everywhere
{i in {i in [1,100]|i%2==0} | i =in {i in {i in [1,100] | i%3==0} | i%5==0} };

function ShrinkSet(Set,i,j,k) 
  return {i in {i in [1,j]|i%k==0} | i =in {j in {k in Set | k%3==0} | j%k==0} };
  end;
ShrinkSet([1,100],10,200,5);