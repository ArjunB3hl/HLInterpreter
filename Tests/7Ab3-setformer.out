{}
{1}
{1, 2, 3, 4}
{}
{3, 6, 9}
{11, 22, 33, 44, 55, 66, 77, 88, 99}
{2, 4}
{6, 12, 18, 24, 30, 36, 42, 48, 54, 60, 66, 72, 78, 84, 90, 96}
{6, 12, 18, 24, 30, 36, 42, 48, 54, 60, 66, 72, 78, 84, 90, 96}
{11, 22, 33, 44, 55, 66, 77, 88, 99}
{15, 30, 45, 60, 75, 90}
{15, 30, 45, 60, 75, 90}
{30, 60, 90}
{15, 30, 45, 60, 75, 90}
